scripts/build/types.js

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env node
+
+/* eslint-disable import/no-commonjs, no-console */
+
+const fs = require('fs');
+const path = require('path');
+const shell = require('shelljs');
+
+shell.exec(
+  'tsc -p tsconfig.declaration.json --outDir temp/types; mv temp/types/index.es.d.ts temp/types/index.d.ts'
```

Comment:
you need to create the temp folder probably first?

Reply:
could we not use something like `require('os').tmpdir()` to store this instead, wdyt?

Reply:
> you need to create the temp folder probably first?

No , it will be created on that command if do not exist.

> require('os').tmpdir()

Not sure to trust in this global command in all environments.

Reply:
the command is specifically  for cross-platform, but if typescript creates the directory, it's fine